Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -6112,4 +6119,10 @@ bool RenderBox::overflowChangesMayAffectLayout() const
     return !canUseOverlayScrollbars() && (style().overflowY() == Overflow::Auto || style().overflowX() == Overflow::Auto);
 }
 
+bool RenderBox::overflowsContainingBlock() const
```

Comment:
This name reads like a generic function but it applies specific box edges and it's hard to tell at what point it is okay to call (maybe turn it into a static helper where these constraints are implicitly met?). don't feel too strongly but it seems like a function that is easy to misuse.

Reply:
Made it a standalone function in AnchorPositiionEvaluator